corehq/apps/export/export.py

```diff
@@ -128,18 +128,26 @@ def get_export_file(export_instances, filters, progress_tracker=None):
     with writer.open(export_instances):
         for export_instance in export_instances:
             # TODO: Don't get the docs multiple times if you don't have to
-            docs = _get_export_documents(export_instance, filters)
+            docs = _get_export_documents_scroll(export_instance, filters)
             _write_export_instance(writer, export_instance, docs, progress_tracker)
 
     return ExportFile(writer.path, writer.format)
 
 
-def _get_export_documents(export_instance, filters):
+def _get_export_documents_scroll(export_instance, filters):
```

Comment:
https://github.com/dimagi/commcare-hq/pull/12438/commits/a6327433d57b1d34403640d5bb9e23fe6fb25bf3 Changed it back to one function. I think it still requires two queries though

https://github.com/dimagi/commcare-hq/pull/12438/files#diff-1086f45b5d86ddf7a9cd7b848caf29a6R1647 We check for errors/permissions before sending the task to celery